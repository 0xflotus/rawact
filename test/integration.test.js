const fs = require("fs");
const path = require("path");
const babel = require("@babel/core");

const testCasesPath = path.resolve(__dirname, "cases");

describe("integration", () => {
	const testCases = fs
		.readdirSync(testCasesPath)
		.filter(file => file.endsWith(".js"));
	for (const filename of testCases) {
		const name = path.basename(filename, path.extname(filename));
		it(`should compile ${name}`, () => {
			const casePath = path.resolve(testCasesPath, filename);
			const input = fs.readFileSync(casePath);
			const { code } = babel.transformFileSync(casePath, {
				caller: {
					name: "test-cases",
					supportsStaticESM: false
				},
				filename: casePath,
				babelrc: false,
				presets: ["@babel/preset-react"],
				plugins: [babel.createConfigItem(require("../"))]
			});
			expect(code).toMatchSnapshot();
		});
	}
});
