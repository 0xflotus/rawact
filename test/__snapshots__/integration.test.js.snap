// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration should compile basic 1`] = `
"import _rawact_createElement from \\"babel-plugin-rawact/lib/runtime/createElement\\";
import _rawact_renderInternal from \\"babel-plugin-rawact/lib/runtime/renderInternal\\";
import _rawact_unmountInternal from \\"babel-plugin-rawact/lib/runtime/unmountInternal\\";
import _rawact_renderChildren from \\"babel-plugin-rawact/lib/runtime/renderChildren\\";
import _rawact_hooks from \\"babel-plugin-rawact/lib/runtime/hooks\\";
const _instructions = {};
const _instructions2 = {};
import * as React from \\"babel-plugin-rawact/lib/runtime/react\\";

const Hello = ({
  name
}) => (() => {
  const _child = name;
  return _context => {
    if (_context._ !== _instructions) {
      if (_context.$) _context.$();

      _context.$ = () => {
        _rawact_unmountInternal(_context.b_);
      };

      _context._ = _instructions;
      _context.a = _rawact_createElement(\\"p\\");

      _rawact_renderInternal(_context, _child, \\"b\\", 1);

      _rawact_renderChildren(_context.a, [\\"Hello \\", _context[\\"b\\"], \\"!\\"]);
    } else {
      _rawact_renderInternal(_context, _child, \\"b\\", 0);
    }

    return _context.a;
  };
})();

export default (() => {
  const _child2 = _rawact_hooks(Hello, {
    name: \\"World\\"
  });

  return _context2 => {
    if (_context2._ !== _instructions2) {
      if (_context2.$) _context2.$();

      _context2.$ = () => {
        _rawact_unmountInternal(_context2.b_);
      };

      _context2._ = _instructions2;
      _context2.a = _rawact_createElement(\\"div\\");
      _context2.a.title = \\"hello\\";

      _rawact_renderInternal(_context2, _child2, \\"b\\", 1);

      _rawact_renderChildren(_context2.a, [_context2[\\"b\\"], \\"!!\\"]);
    } else {
      _rawact_renderInternal(_context2, _child2, \\"b\\", 0);
    }

    return _context2.a;
  };
});"
`;
